### PR TITLE
Neural Coder use INC API by default instead of Optimum

### PR DESCRIPTION
**Pull request by kaiyaointel**
## Type of Change

change logic of which default API design to be enabled

## Description

1. a change of default adopted optimization API's logic:
 
- Old logic:
-- Default all use Optimum API for launchers, no matter INC launcher or Optimum launcher
- New logic:
-- If user use INC launcher, then default API is INC API. 
-- If user use Optimum launcher, then default API is Optimum API.

Optimum side's modification is in this [PR](https://github.com/huggingface/optimum-intel/pull/183)

2. also refined some backend yamls

3. also added pre-requisite in Python Launcher example

## How has this PR been tested?

locally

## Dependency Change?

no

### CODE REVIEW

**Comment by kaiyaointel** [comment]
@chensuyue hi, modification is done: in the future INC launcher will enable INC API, Optimum launcher will enable Optimum API. Optimum PR is https://github.com/huggingface/optimum-intel/pull/183

File: neural_coder/docs/PythonLauncher.md [hunk 1]
```diff
@@ -5,7 +5,14 @@ Neural Coder can be used as a Python **Launcher**. Users can run the Python mode
 
 ## Quick-Start
 
-Example: Let's say you are running an NLP model using ```run_glue.py``` from HuggingFace transformers [examples](https://github.com/huggingface/transformers/blob/v4.21-release/examples/pytorch/text-classification/run_glue.py), you generally run with a Python command line like this:
+Example: Let's say you are running an NLP model using ```run_glue.py``` from HuggingFace transformers [examples](https://github.com/huggingface/transformers/blob/v4.21-release/examples/pytorch/text-classification/run_glue.py).
+
+Pre-requisites:
+```bash
+pip install transformers==4.21.0 torch datasets
```
Comment (chensuyue) [inline neural_coder/docs/PythonLauncher.md:12]: How about adding `pip install neural-compressor` ?

File: neural_coder/backends/pytorch_inc_dynamic_quant.yaml [hunk 2]
```diff
@@ -17,30 +17,29 @@ transformation:
...
-      [+]         pass
+      [+] def eval_func(model):
+      [+]     EVAL_FUNC_LINES
+      [+] try:
+      [+]     torch.backends.quantized.engine = 'onednn'
+      [+] except:
+      [+]     from torch.backends.quantized import engine; engine = 'onednn'
+      [+] from neural_compressor.conf.config import QuantConf
+      [+] from neural_compressor.experimental import Quantization, common
+      [+] quant_config = QuantConf()
```
Comment (chensuyue) [inline neural_coder/backends/pytorch_inc_dynamic_quant.yaml:28]: Seems it's still using 1.x API, when will Neural Coder migrate into 2.0 API?

**Comment by chensuyue** [comment]
Local test pass, will merge first, please check the comments.

File: neural_coder/docs/PythonLauncher.md (hunk 1 above)
Comment (kaiyaointel) [reply neural_coder/docs/PythonLauncher.md:12]: sure

File: neural_coder/backends/pytorch_inc_dynamic_quant.yaml (hunk 2 above)
Comment (kaiyaointel) [reply neural_coder/backends/pytorch_inc_dynamic_quant.yaml:28]: I don't know about the high-level plan, from technical perspective, if INC 2.0 API is ready, then migration is doable at any time

**Pull request merged**